src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java

```diff
@@ -89,6 +90,10 @@ public Response createDataAccessRequest(
       @Auth AuthUser authUser, @Context UriInfo info, String dar) {
     try {
       User user = findUserByEmail(authUser.getEmail());
+      if (Objects.isNull(user.getLibraryCards()) || user.getLibraryCards().isEmpty()) {
+        throw new IllegalArgumentException("User must have a library card to create a DAR.");
```

Comment:
I think a `NotAllowedException` is more appropriate here.

Reply:
I think NotAllowedException is when the user does the wrong method - e.g. a GET on an endpoint that just has PUT. E.g., this example I found online:
``` java
public Response handle() {
  if (HttpMethod.GET.equals(request.getMethod())) {
       // logic
   }
  throw new NotAllowedException("GET"); // only GET is allowed.
}
```